python/bin/p1_lband_extract

```diff
@@ -0,0 +1,147 @@
+    log_parser = parser.add_argument_group('Log Control')
+    log_parser.add_argument(
+        '--ignore-index', action='store_true',
+        help="If set, do not load the .p1i index file corresponding with the .p1log data file. If specified and a .p1i "
+             "file does not exist, do not generate one. Otherwise, a .p1i file will be created automatically to "
+             "improve data read speed in the future.")
+    log_parser.add_argument(
+        '--log-base-dir', metavar='DIR', default=DEFAULT_LOG_BASE_DIR,
+        help="The base directory containing FusionEngine logs to be searched if a log pattern is specified.")
+    log_parser.add_argument(
+        '--progress', action='store_true',
+        help="Print file read progress to the console periodically.")
+    log_parser.add_argument(
+        'log',
+        help="The log to be read. May be one of:\n"
+             "- The path to a .p1log file or a file containing FusionEngine messages and other content\n"
+             "- The path to a FusionEngine log directory\n"
+             "- A pattern matching a FusionEngine log directory under the specified base directory "
+             "(see find_fusion_engine_log() and --log-base-dir)")
+
+    options = parser.parse_args()
+
+    read_index = not options.ignore_index
+    generate_index = not options.ignore_index
+
+    # Configure logging.
+    if options.verbose >= 1:
+        logging.basicConfig(level=logging.INFO, format='%(levelname)s - %(name)s:%(lineno)d - %(message)s',
+                            stream=sys.stdout)
+        if options.verbose == 1:
+            logging.getLogger('point_one.fusion_engine.parsers').setLevel(logging.DEBUG)
+        else:
+            logging.getLogger('point_one.fusion_engine.parsers').setLevel(
+                logging.getTraceLevel(depth=options.verbose - 1))
+    else:
+        logging.basicConfig(level=logging.INFO, format='%(message)s', stream=sys.stdout)
+
+    HighlightFormatter.install(color=True, standoff_level=logging.WARNING)
+    BrokenPipeStreamHandler.install()
+
+    # Locate the input file and set the output directory.
+    input_path, output_dir, log_id = locate_log(input_path=options.log, log_base_dir=options.log_base_dir, return_log_id=True, return_output_dir=True,
+                                                extract_fusion_engine_data=False)
+    if input_path is None:
+        # locate_log() will log an error.
+        sys.exit(1)
+
+    if options.output is not None:
+        output_dir = options.output
+
+    # Read through the data file, searching for valid LBandFrame messages to extract and store in
+    # 'output_dir/<prefix>.p1log'.
+    if options.prefix is not None:
+        prefix = options.prefix
+    elif log_id is not None:
+        prefix = 'lband'
+    else:
+        prefix = os.path.splitext(os.path.basename(input_path))[0]
+    output_path = os.path.join(output_dir, prefix + '.rtcm')
```

Comment:
L-band data is not guaranteed to be RTCM, so we probably should not use this file extension. Could call it `.lband` but that's a bit weird. If we use a generic extension like `.bin` or something, we might want to call the file `lband.bin` instead of `<prefix>.bin` since that would be too generic.

Or we just have the user specify the output filename at the command line.